src/libs/actions/Report.js

```diff
@@ -1084,6 +1094,81 @@ function updateLastReadActionID(reportID, sequenceNumber, manuallyMarked = false
     });
 }
 
+/**
+ * Gets the latest page of report actions and updates the last read message
+ *
+ * @param {Number} reportID
+ */
+function openReport(reportID) {
+    const sequenceNumber = reportMaxSequenceNumbers[reportID];
+    lastReadSequenceNumbers[reportID] = sequenceNumber;
+    API.write('OpenReport',
+        {
+            reportID,
+            sequenceNumber,
+        },
+        {
+            optimisticData: [{
+                onyxMethod: 'merge',
+                key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+                value: {
+                    lastVisitedTimestamp: Date.now(),
+                    unreadActionCount: getUnreadActionCountFromSequenceNumber(reportID, sequenceNumber),
+                },
+            }],
+        });
+}
+
+/**
+ * Marks the new report actions as read
+ *
+ * @param {Number} reportID
+ */
+function readNewestAction(reportID) {
+    const sequenceNumber = reportMaxSequenceNumbers[reportID];
+    lastReadSequenceNumbers[reportID] = sequenceNumber;
+    API.write('ReadNewestAction',
+        {
+            reportID,
+            sequenceNumber,
+        },
+        {
+            optimisticData: [{
+                onyxMethod: 'merge',
+                key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+                value: {
+                    lastVisitedTimestamp: Date.now(),
+                    unreadActionCount: getUnreadActionCountFromSequenceNumber(reportID, sequenceNumber),
+                },
+            }],
+        });
+}
+
+/**
+ * Sets the last read comment on a report
+ *
+ * @param {Number} reportID
+ * @param {Number} sequenceNumber
+ */
+function markCommentAsUnread(reportID, sequenceNumber) {
+    lastReadSequenceNumbers[reportID] = sequenceNumber;
+    API.write('MarkCommentAsUnread',
+        {
+            reportID,
+            sequenceNumber,
+        },
+        {
+            optimisticData: [{
+                onyxMethod: 'merge',
+                key: `${ONYXKEYS.COLLECTION.REPORT}${reportID}`,
+                value: {
+                    lastVisitedTimestamp: Math.round(Date.now() / 1000),
```

Comment:
NAB and sorry, as mentioned in that other comment I was wrong about this and `Date.now()` is totally fine to use as it returns milliseconds and the lastVisitedTimestamp should be in `ms`.